llmfoundry/optim/lion.py

```diff
@@ -103,14 +106,14 @@ def step(self, closure: Optional[Callable] = None):
 
         return loss
 
-    def dist_reduce_metrics(self, optimizer_metrics):
+    def dist_reduce_metrics(self, optimizer_metrics: Dict[str, torch.Tensor]):
         for metric in optimizer_metrics:
             if metric.startswith('l2_norm'):
                 reduced = optimizer_metrics[metric]
                 if dist.get_world_size() > 1:
                     dist.all_reduce(reduced, reduce_operation='SUM')
 
-                optimizer_metrics[metric] = math.sqrt(reduced)
+                optimizer_metrics[metric] = torch.tensor(math.sqrt(reduced))
```

Comment:
is changing the type here ok?

Reply:
Yeah, they're all supposed to be tensors, this is the one exception. The code that uses it checks if it's a tensor and then calls .item if it is